### DISCUSSION TITLE
Custom move and copy command works in terminal but not in lf subshell

### DISCUSSION THREAD

**Discussion by jan0sch** [gokcehan/lf#1574] [Q&A] [answered]
Hi,

I followed some instructions to setup custom copy and move commands to preserve file attributes (like timestamps) with `rsync` and `mv`. However on some files these operations are not executed and lf does never display an error. Some digging found that while `rsync` (for copy) seems robust for most of the cases `mv` is more picky. So I adjusted the code and have now something like this:

```
cmd paste &{{
  while read -r line; do
    set -- "$@" "$(printf '%q' $line)"
  done < ~/.local/share/lf/files
  mode="$1"
  shift
  case "$mode" in
    copy)
      echo "lf-paste in ${PWD}: rsync -av --backup -- $@ ." >> ~/.cache/lf.log
      rsync -av --backup --progress -- "$@" . |
      stdbuf -i0 -o0 -e0 tr '\r' '\n' |
      while IFS= read -r line; do
        lf -remote "send $id echo $line"
      done
      ;;
    move)
      echo "lf-paste in ${PWD}: mv -n -- $@ ." >> ~/.cache/lf.log
      mv -n -- $@ .;;
  esac
  rm ~/.local/share/lf/files
  lf -remote "send clear"
}}
```

But this still does not work, especially if files or directories contain _special_ characters like spaces. However the arguments are escaped properly and I can simply copy the command from my made-up logfile and execute it and it will work just fine. But it does not from the subshell of lf. Debugging there brings error messages like:

```
mv: rename /some/path/with/escaped\ spaces/file.txt to ./file.txt: No such file or directory
```

The line from my log will say something like this:

```
lf-paste in some-directory: mv -n -- /some/path/with/escaped\ spaces/file.txt .
```

and I can copy the `mv` command from there paste it into a terminal and it will run without problems. I already tried quoting `$@` (i.e. tried `"$@"` and `$@`) but that didn't have any effect.

As mentioned `rsync` works nearly all of the time but there I've also had this behaviour occasionally.

Any ideas what could cause this issue?

lf: 31
go: 1.20
shell: zsh 5.9
os: freebsd 14

Kind regards

**Comment by joelim-work** [comment, marked as answer]
I think it's because `$@` is already an array (where elements can contain spaces), so there is no need to use `"$(printf '%q' $line)"`. Do soing ends up adding a literal backslash into the array element and results in a 'no such file' error because now the filename actually contains `\`.

I changed the line at the top to just `set -- "$@" "$line"` and the script works for me.

**Reply by jan0sch** [reply]
Awesome, thanks! That really did it. :heart:
